src/sentry/auth/staff.py

```diff
@@ -0,0 +1,318 @@
+from __future__ import annotations
+
+import ipaddress
+import logging
+from datetime import datetime, timedelta
+
+from django.conf import settings
+from django.core.signing import BadSignature
+from django.utils import timezone
+from django.utils.crypto import constant_time_compare, get_random_string
+from rest_framework.request import Request
+
+from sentry.auth.elevated_mode import ElevatedMode
+from sentry.auth.system import is_system_auth
+from sentry.utils.auth import has_completed_sso
+
+logger = logging.getLogger("sentry.staff")
+
+SESSION_KEY = "_stf"
+
+COOKIE_NAME = getattr(settings, "STAFF_COOKIE_NAME", "stf")
+
+COOKIE_SALT = getattr(settings, "STAFF_COOKIE_SALT", "")
+
+COOKIE_SECURE = getattr(settings, "STAFF_COOKIE_SECURE", settings.SESSION_COOKIE_SECURE)
+
+COOKIE_DOMAIN = getattr(settings, "STAFF_COOKIE_DOMAIN", settings.SESSION_COOKIE_DOMAIN)
+
+COOKIE_PATH = getattr(settings, "STAFF_COOKIE_PATH", settings.SESSION_COOKIE_PATH)
+
+COOKIE_HTTPONLY = getattr(settings, "STAFF_COOKIE_HTTPONLY", True)
+
+# the maximum time the session can stay alive
+MAX_STAFF_SESSION_AGE = timedelta(hours=4)
+
+IDLE_MAX_STAFF_SESSION_AGE = timedelta(minutes=15)
```

Comment:
we can do this without fetching from settings because `Staff` doesn't exist on self-hosted, right?

otherwise we get these from `settings` for superuser because they can be customized on self-hosted? 🤔

Reply:
Good point, I didn't think of self-hosted and removed it originally because I didn't see the settings anywhere in the codebase, but they were prob put there for self-hosted.

Reply:
Good point, I didn't think of self-hosted and removed it originally because I didn't see the settings anywhere in the codebase, but they were prob put there for self-hosted.

Reply:
i think it's ok the way it is -- i just wanted to make sure i'm understanding correctly